### PR TITLE
fix: limit the UTXOs of the miners to avoid bug

### PR DESCRIPTION
**Pull request by obycode**
See https://github.com/stacks-network/stacks-core/issues/5224

### CODE REVIEW

**Review by deantchi — APPROVED** [review]
I'm less familiar with this repo, but given how the rest of the script is written, I think it should work.

File: docker-compose.yml [hunk 1]
```diff
@@ -83,7 +83,14 @@ services:
         bitcoin-cli -rpcconnect=bitcoind importaddress $${BTC_ADDR} "" false
         bitcoin-cli -rpcconnect=bitcoind generatetoaddress $${INIT_BLOCKS} $${BTC_ADDR}
         DEFAULT_TIMEOUT=$$(($$(date +%s) + 30))
+        RANDOM_ADDR=$$(bitcoin-cli -rpcconnect=bitcoind getnewaddress)
```
[the remark this answers is not in the document]
Comment (obycode) [reply docker-compose.yml:86]: Isn't this only called once? It's outside of the `while` loop.

**Comment by obycode** [comment]
I'm not clear why [build-stacks-node](https://github.com/hirosystems/stacks-regtest-env/pull/45#logs) is failing. Is it the Rust version? This works locally.

**Review by zone117x — APPROVED** [review]
Good hack :D

**Comment by obycode** [comment]
@janniks are you available to look into why the build fails here?

**Comment by janniks** [comment]
I ran into these before. We can use a newer blockchain commit or add a `cargo update` in the dockerfile (works for me locally, but not sure if it's a good practice)

**Review by zone117x — APPROVED** [review]

**Comment by zone117x** [comment]
@obycode should this PR be closed in favor of https://github.com/hirosystems/stacks-regtest-env/pull/47 ? Otherwise there's a bunch of merge conflicts that need fixed in the mining bash script.

**Comment by obycode** [comment]
Yup, we can close this one.

**Pull request closed without merging**